deps/rabbit/src/rabbit_peer_discovery.erl

```diff
@@ -82,6 +85,16 @@ node_type() ->
             ?DEFAULT_NODE_TYPE
     end.
 
+-spec registration() -> true | false.
+
+registration() ->
+    case application:get_env(rabbit, cluster_formation) of
+        {ok, Proplist} ->
+            proplists:get_value(registration, Proplist, ?DEFAULT_REGISTRATION);
```

Comment:
If the value is set to something else that a list, this code will crash.

You can do something like:
```erlang
    case application:get_env(rabbit, cluster_formation) of
        {ok, Proplist} when is_list(Proplist) ->
            proplists:get_value(registration, Proplist, ?DEFAULT_REGISTRATION);
        {ok, Other} ->
            ?LOG_ERROR(
               "Peer discovery: ignoring invalid value for `register`: ~0p",
               [Other],
               #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
            ?DEFAULT_REGISTRATION;
```

Reply:
Also, you use the name `registration` but the schema uses `rabbit.cluster_formation.register`. I think there is an inconsistency here.